### PR TITLE
Squad inference example

### PR DESCRIPTION
**Pull request by yzhang123**

### CODE REVIEW

File: nemo/collections/nlp/nm/data_layers/qa_squad_datalayer.py [hunk 1]
```diff
@@ -59,9 +59,9 @@ def output_ports(self):
             "input_ids": NeuralType(('B', 'T'), ChannelType()),
             "input_type_ids": NeuralType(('B', 'T'), ChannelType()),
             "input_mask": NeuralType(('B', 'T'), ChannelType()),
-            "start_positions": NeuralType(tuple('B'), ChannelType()),
-            "end_positions": NeuralType(tuple('B'), ChannelType()),
             "unique_ids": NeuralType(tuple('B'), ChannelType()),
+            "start_positions": NeuralType(tuple('B'), ChannelType(), optional=True),
+            "end_positions": NeuralType(tuple('B'), ChannelType(), optional=True),
```
Comment (blisc) [inline nemo/collections/nlp/nm/data_layers/qa_squad_datalayer.py]: @okuchaiev What exactly does optional mean? How does it work if I only return end_positions but not start_positions?

Comment (yzhang123) [reply nemo/collections/nlp/nm/data_layers/qa_squad_datalayer.py]: start_positions, end_positions are the "labels", they are optional to make the data_layer more general for inference cases where i do not have the groundtruth.

Comment (blisc) [reply nemo/collections/nlp/nm/data_layers/qa_squad_datalayer.py]: That makes sense, I just didn't know we had this option, and want to know how it works.

File: examples/nlp/question_answering/question_answering_squad_infer.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+        start_logits=start_logits,
+        end_logits=end_logits,
+        n_best_size=args.n_best_size,
+        max_answer_length=args.max_answer_length,
+        version_2_with_negative=args.version_2_with_negative,
+        null_score_diff_threshold=args.null_score_diff_threshold,
+        do_lower_case=args.do_lower_case,
+    )
+
+    if args.output_prediction_file is not None:
```
Comment (ekmb) [inline examples/nlp/question_answering/question_answering_squad_infer.py]: can we make with argument required? there is no else

File: examples/nlp/question_answering/question_answering_squad_infer.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+        type=str,
+        required=False,
+        default="predictions.json",
+        help="File to write predictions to. Only in evaluation mode.",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def create_pipeline(
```
Comment (ekmb) [inline examples/nlp/question_answering/question_answering_squad_infer.py]: if you're reusing the same pipeline, would it be simpler to add inference to the existing example?

Comment (yzhang123) [reply examples/nlp/question_answering/question_answering_squad_infer.py]: makes sense

File: examples/nlp/question_answering/question_answering_squad_infer.py (hunk 2 above)
Comment (yzhang123) [reply examples/nlp/question_answering/question_answering_squad_infer.py]: If the user is not interested in the prediction e.g. for training, its better not the save a file to the users work space.
